deliver/lib/deliver/submit_for_review.rb

```diff
@@ -58,7 +58,7 @@ def wait_for_build(app)
       build = nil
 
       loop do
-        latest_build = find_build(app.latest_version.candidate_builds)
+        latest_build = find_build(app.latest_version)
```

Comment:
`find_build` makes this have a loop in a loop which feels a bit too loopy 😊 Is it possible to reuse this same loop here?

Maybe we simply this and just do something like...
```rb
loop do
  # Waits for candidate builds to appear
  candidate_builds = app.latest_version.candidate_builds
  if (candidate_builds || []).count == 0
    UI.message("Waiting for candidate builds to appear...")
    unless (Time.now - start) > (60 * 5)
      next
    else
      UI.user_error!("Could not find any available candidate builds on App Store Connect to submit")
    end
  end
```

Reply:
@joshdholtz that totally makes sense, I'll update and test the code as soon as I can!

Reply:
@mjovkovic @joshdholtz 
This fix is for deliver only. 
We have seen few issues in pilot.
Can this be a fix for pilot as well ?